flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/ResultPartitionWriter.java

```diff
@@ -35,23 +35,23 @@
  * In this case {@link ResultPartitionWriter#fail(Throwable)} still needs to be called afterwards to fully release
  * all resources associated the the partition and propagate failure cause to the consumer if possible.
  */
-public interface ResultPartitionWriter extends AutoCloseable {
+public abstract class ResultPartitionWriter implements AutoCloseable, AvailabilityProvider {
```

Comment:
The interface can only extend another interface in this case, so we have to redefine the duplicate methods `#isAvailable()` and `#getAvailableFuture()` for `ResultPartitionWriter`. If we change it to an abstract class, then it can implement multiple other interfaces including existing `AvailabilityProvider`. I would refactor this commit message a bit to make it more clearly.

Reply:
I think there are no issues with an interface extending multiple interfaces.

Reply:
Yes, it is no issue here, and my memory was wrong.